regression/esbmc-solidity/polymorphism_1/polymorphism.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-3.0
+
+// Different contracts with the same function name. 
+pragma solidity >=0.5.0;
+
+contract A {
+
+    function poly() public pure returns (uint8) {
+        return 21;
+    }
+}
+
+contract B {
+    function poly() public pure returns (uint8) {
+        return 42;
+    }
+}
+
+contract C {
+    function test_poly() public {
+        A contract_A = new A();
+        B contract_B = new B();
+        assert(contract_A.poly() == 21);
```

Comment:
May I ask if the verification fails if we change this assertion to `assert(contract_A.poly() == 42);`

Reply:
Yes, the verification failed with counterexample given:
```
State 1 file polymorphism.sol line 23 function test_poly thread 0
----------------------------------------------------
Violated property:
  file polymorphism.sol line 23 function test_poly
  assertion
  (signed int)return_value$_poly$3 == 42


VERIFICATION FAILED
```